libs/@local/graph/postgres-store/src/store/postgres/ontology/entity_type.rs

```diff
@@ -1753,40 +1753,79 @@ where
+                    .map(|metadata| metadata.id.clone())
+                    .collect::<HashSet<_>>(),
+            );
+        }
+
+        let types_and_parents = entity_type_id_set
+            .iter()
+            .flat_map(|(base, parents)| iter::once(base.clone()).chain(parents.clone()))
+            .collect::<HashSet<_>>()
+            .into_iter()
+            .collect::<Vec<_>>();
+
+        self.build_entity_type_context(&types_and_parents, &mut policy_context_builder)
             .await
             .change_context(QueryError)?;
-
         let policy_context = policy_context_builder.build().change_context(QueryError)?;
 
-        entity_type_ids
-            .iter()
-            .map(|entity_type_id| {
-                policy_set
-                    .evaluate(
-                        &Request {
-                            actor: Some(actor),
-                            action: ActionName::Instantiate,
-                            resource: Some(&PartialResourceId::EntityType(Some(Cow::Borrowed(
-                                entity_type_id.into(),
-                            )))),
-                            context: RequestContext::default(),
-                        },
-                        &policy_context,
-                    )
-                    .change_context(QueryError)
-                    .map(|authorized| match authorized {
-                        Authorized::Always => true,
-                        Authorized::Never => false,
-                        Authorized::Partial(partial) => {
-                            tracing::error!(
-                                "Instantiation checking is not supported for partial \
-                                 authorization:\n
+        entity_type_id_set
+            .into_iter()
+            .map(|(base, parents)| {
+                // We need to check the base entity type and all its parents
+                // to see if the user can instantiate it.
+                for entity_type_id in iter::once(base).chain(parents) {
+                    let allowed = policy_set
+                        .evaluate(
+                            &Request {
+                                actor: Some(actor),
+                                action: ActionName::Instantiate,
+                                resource: Some(&PartialResourceId::EntityType(Some(Cow::Owned(
+                                    EntityTypeId::new(entity_type_id),
+                                )))),
+                                context: RequestContext::default(),
+                            },
+                            &policy_context,
+                        )
+                        .change_context(QueryError)
+                        .map(|authorized| match authorized {
+                            Authorized::Always => true,
+                            Authorized::Never => false,
+                            Authorized::Partial(partial) => {
+                                tracing::error!(
```

Comment:
shouldn't this error out instead of failing silently?

Reply:
This is actually something which should never happen because we define an ID above. I could make it an `unreachable!` instead

Reply:
757ac38b05dd520a21f360a684fb686abb3b2f7a